test/sys/test_ptrace.rs

```diff
@@ -273,3 +273,187 @@ fn test_ptrace_syscall() {
         }
     }
 }
+
+#[test]
+#[cfg(all(
+    target_os = "linux",
+    all(target_arch = "x86_64", any(target_env = "gnu", target_env = "musl")),
+))]
+fn test_ptrace_setregs() {
+    use nix::sys::ptrace;
+    use nix::sys::signal::{raise, Signal};
+    use nix::sys::wait::{waitpid, WaitPidFlag, WaitStatus};
+    use nix::unistd::fork;
+    use nix::unistd::ForkResult::*;
+
+    require_capability!("test_ptrace_setregs", CAP_SYS_PTRACE);
+
+    let _m = crate::FORK_MTX.lock();
+    match unsafe { fork() }.expect("Error: Fork Failed") {
+        Child => {
+            ptrace::traceme().unwrap();
+            // As recommended by ptrace(2), raise SIGTRAP to pause the child
+            // until the parent is ready to continue
+            loop {
+                raise(Signal::SIGTRAP).unwrap();
+            }
+        }
+        Parent { child } => {
+            assert_eq!(
+                waitpid(child, None),
+                Ok(WaitStatus::Stopped(child, Signal::SIGTRAP))
+            );
+
+            let mut regs = ptrace::getregs(child).expect("Failed to get regs.");
+            regs.rax = 0x42424242;
```

Comment:
AFAIK, this register name is the only difference between the x86 and x86_64 tests.  Could you combine them to reduce code duplication?  Even better if you can combine them with the setfpregs test too.